src/components/sdk/MeetingControls/AudioInputVFControl.tsx

```diff
@@ -0,0 +1,140 @@
+import { useToggleLocalMute } from '../../../hooks/sdk/useToggleLocalMute';
+
+interface Props {
+  /** The label that will be shown when microphone is muted, it defaults to `Mute`. */
+  muteLabel?: string;
+  /** The label that will be shown when microphone is unmuted, it defaults to `Unmute`. */
+  unmuteLabel?: string;
+  /** Title attribute for the icon when muted, it defaults to `Muted microphone` */
+  mutedIconTitle?: string;
+  /** Title attribute for the icon when unmuted, it defaults to `Microphone` */
+  unmutedIconTitle?: string;
+}
+
+const AudioInputVFControl: React.FC<Props> = ({
+  muteLabel = 'Mute',
+  unmuteLabel = 'Unmute',
+  mutedIconTitle,
+  unmutedIconTitle
+}) => {
+  const meetingManager = useMeetingManager();
+  const [voiceFocusOn, setVoiceFocusOn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [enableVoiceFocus, setEnableVoiceFocus] = useState(false);
+  const dropdownWithVFOptions = useRef<PopOverItemProps[] | null>(null);
+  const { muted, toggleMute } = useToggleLocalMute();
+  const [device, setDevice] = useState<Device | AudioTransformDevice | null>(meetingManager.selectedAudioInputTransformDevice);
+  const { supportsVoiceFocus, addVoiceFocus } = useVoiceFocus();
+  const audioInputConfig: DeviceConfig = {
+    additionalDevices: true,
+  };
+  const { devices, selectedDevice } = useAudioInputs(audioInputConfig);
+
+  const dropdownOptions: PopOverItemProps[] = devices.map((device) => ({
+    children: <span>{device.label}</span>,
+    checked: isOptionActive(selectedDevice, device.deviceId),
+    onClick: (): Promise<void> =>
+      meetingManager.selectAudioInputDevice(device.deviceId),
+  }));
+
+  useEffect(() => {
+    if (device instanceof VoiceFocusTransformDevice) {
+      setEnableVoiceFocus(true);
+    } else {
+      setEnableVoiceFocus(false);
+    }
+  }, [device]);
+
+  useEffect(() => {
+    meetingManager.subscribeToSelectedAudioInputTransformDevice(setDevice);
+
+    return (): void => {
+      meetingManager.unsubscribeFromSelectedAudioInputTransformDevice(setDevice);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (supportsVoiceFocus) {
+      const vfOption: PopOverItemProps = {
+        children: <><PopOverSeparator />{isLoading && <Spinner />} {enableVoiceFocus ? <span>Disable voice focus</span> : <span>Enable voice focus</span>}</>,
+        checked: enableVoiceFocus,
+        disabled: isLoading,
+        onClick: () => {
+          setVoiceFocusOn(current => !current);
+        },
+      };
+      dropdownOptions?.push(vfOption);
+    }
+    dropdownWithVFOptions.current = dropdownOptions;
+  }, [supportsVoiceFocus, enableVoiceFocus, voiceFocusOn, isLoading, dropdownOptions]);
+
+  useEffect(() => {
+    async function onVFCheckboxChange() {
+      let current;
+      setIsLoading(true);
+      if (voiceFocusOn) {
+        if (typeof (device) === 'string') {
+          const currentDevice = audioInputSelectionToDevice(device);
```

Comment:
The `device` passed here can be 'None' hence currentDevice will be `null` then, have not checked the `addVoiceFocus` yet but checking if it will support it?

Reply:
Added the checker in the disable props on the voice focus option.

Reply:
Just checked the meetingV2 demo, seems like we can take the `null` as the input device for voice focus transformer. Removed the checker